tgg/annotations/get_ensembl_db_info.py

```diff
@@ -22,53 +24,168 @@
 
+    return gene_id_to_transcript_id
 
-def get_canonical_transcript_created_modified_dates(database=CURRENT_ENSEMBL_DATABASE):
-    transcript_id_to_dates = {}
-    with pymysql.connect(host="useastdb.ensembl.org", user="anonymous", database=database) as conn:
-        with conn.cursor() as c:
-            columns = ["transcript.stable_id", "transcript.created_date", "transcript.modified_date"]
-            columns_str = ", ".join(columns)
-            c.execute(f"SELECT {columns_str} FROM gene LEFT JOIN transcript ON canonical_transcript_id = transcript_id")
 
-            for row in c:
-                d = dict(zip(columns, row))
-                gene_id = d['transcript.stable_id']
-                transcript_id_to_dates[gene_id] = d['transcript.created_date'], d['transcript.modified_date']
+def get_gene_id_to_transcript_ids(
+        database=CURRENT_ENSEMBL_DATABASE,
+        only_protein_coding=False,
+        only_canonical_transcripts=False):
+    """Returns a dictionary mapping each Ensembl gene_id => a list of transcript ids for that gene.
+
+    Args:
+        database (str): The Ensembl database name (eg. "homo_sapiens_core_107_38")
+        only_protein_coding (bool): If True, only return protein coding genes
+        only_canonical_transcripts (bool): If True, only return canonical transcripts
+    Return:
+        dict: mapping ENSG id string to a list of ENST id strings
+    """
+
+    gene_id_to_transcript_metadata_list = get_gene_id_to_transcript_metadata(
+        database=database,
+        only_canonical_transcripts=only_canonical_transcripts,
+        only_protein_coding=only_protein_coding)
+
+    return {
+        gene_id: [
+            transcript_metadata["transcript.stable_id"] for transcript_metadata in transcript_metadata_list
+        ]
+        for gene_id, transcript_metadata_list in gene_id_to_transcript_metadata_list.items()
+    }
+
+
+def get_gene_id_to_canonical_transcript_id(database=CURRENT_ENSEMBL_DATABASE, only_protein_coding=False):
+    """Returns a dictionary mapping each Ensembl gene_id => canonical transcript id
+
+    Args:
+        database (str): The Ensembl database name (eg. "homo_sapiens_core_107_38")
+        only_protein_coding (bool): Only include protein-coding genes
+
+    Return:
+        dict: mapping ENSG id string to the canonical ENST id string
+    """
+
+    gene_id_to_transcript_metadata_list = get_gene_id_to_transcript_metadata(
+        database=database,
+        only_canonical_transcripts=True,
+        only_protein_coding=only_protein_coding)
+
+    gene_id_to_canonical_transcript_id = {}
+    for gene_id, transcript_metadata_list in gene_id_to_transcript_metadata_list.items():
+        if len(transcript_metadata_list) > 1:
+            raise Exception(f"{gene_id} has more than 1 canonical transcript")
+        if len(transcript_metadata_list) == 0:
+            raise Exception(f"{gene_id} has 0 canonical transcripts")
```

Comment:
do all non-protein-coding transcripts have canonical transcripts? just curious if this error will be triggered for potentially valid inputs

Reply:
This is a sanity check since the method call above this
```
gene_id_to_transcript_metadata_list = get_gene_id_to_transcript_metadata(
        database=database,
        only_canonical_transcripts=True,
        only_protein_coding=only_protein_coding)
```
should only return genes that do have canonical transcripts (regardless of whether they're coding or not).